### ISSUE TITLE
JavaScript support

### ISSUE THREAD

**Issue by rice7th** [CobaltBSD/neosurf#11]
Nice javascript support would be awesome. At least in my opinion, https://bellard.org/quickjs/ is the best option. Pure, portable C and most importantly, it supports ES2020, meaning Async/await and many nice, modern features are easily available, in contrast to Duktape which supports only the very old ES5 and is very slow compared to quickjs.

Also, not to be too ambitious, but https://github.com/wasm3/wasm3 might be an idea for WASM in the future too ;)

**Comment by CobaltBSD** [comment]
QuickJS support is currently planned; As for WASM3, I will look into it.

**Comment by xplshn** [comment]
What about the SerenityOS JS engine? it seems complete and its fast.

**Comment by xplshn** [comment]
Source and blogpost from developers here https://github.com/SerenityOS/serenity/tree/master/Ladybird

Its got a HaikuOS port so I don't think it'll be so difficult, check the old Ladybird browser repo for build docs.

**Comment by rice7th** [comment]
It isn't optimized as much as quickjs and Quickjs is just much more mature. LibJS is very promising, but not yet complete, while QuickJS supports full Es2020, which is quite hard to accomplish

**Comment by xplshn** [comment]
Please just read the aforementioned blogpost

**Comment by xplshn** [comment]
https://github.com/bellard/quickjs Not updated since 2022, 70 issues open

**Comment by rice7th** [comment]
@xplshn what blog post?

**Comment by xplshn** [comment]
https://github.com/SerenityOS/serenity/tree/master/Ladybird

**Comment by rice7th** [comment]
@xplshn that is not a blog post, however LibJS just got JIT compilation! That is awesome for a js engine that's less than 60KLoC. Also at this point lets use libwasm for wasm, even tho I think wasm3 is still both smaller, faster and more complete, plus it is maintained and written in c99.

**Comment by rice7th** [comment]
@xplshn another update: QuickJS just got updated! The project is **still alive**!

**Comment by patham9** [comment]
@rice7th Yes. I wish the same would be the case for Neosurf too. Seems like it forked from Netsurf and was quickly abandoned after insignificant additions, just a few months after.

**Comment by rice7th** [comment]
@patham9 It's quite sad honestly, but I get it, open source is not really sustainable when you're alone.

**Comment by patham9** [comment]
I disagree @rice7th, it is just a matter of contributing to the actual project properly so that additions and improvements even when rare can accumulate.
Forking it, making small changes, giving it a new name and creating fuzz about it, and then abandoning, is just a counterproductive novice move that also fails to acknowledge the ambitions of the decade-long-ongoing parent project where any actual improvements should have ended up in. In the meanwhile NeoSurf is long obsolete while NetSurf is still in active development.
